### PR TITLE
Sahaj withdrawal pt 107 112

### PR DESCRIPTION
**Pull request by sahajisy**
```
#Author: Sahaj Balgunde
#Keywords Summary : Withdawal Test with Checkbox and Without Checkbox
@WithdrawAccount
Feature: Withdraw user account

Scenario: User is clicking on Back button

Given Login To Platform Portal
Then Click Withdrawal link on Footer
Then Check Back button on Withdrawal Page


Scenario: User is withdrawing the account

Then Click Withdrawal link on Footer
Then Withdraw Account
```

Steps for Account Withdrawal for following test script
1. Create a new account manually
2. Enter newly created login details ie. username and password in WithdrawAccount Sheet.
3. Read those credentials in login scripts.
4. Run the script to withdraw account.

### CODE REVIEW

**Comment by csonparote** [comment]
Write your feature file in your first comment.

File: platform/src/main/java/platform/pageobjects/Withdrawal/Withdrawal.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	@FindBy(xpath = "//a[@class='button button--gray button--medium button--coupon-back']//span[contains(text(),'戻る')]")
+	public WebElement back;
+
+	@FindBy(xpath = "//button[@role='button']")
+	public WebElement withdrawAccount;
+
+	@FindBy(xpath = "//h1[contains(@class,'')]//a[contains(@class,'logo')]")
+	public WebElement topButton;
+
+	public void accountWithdrawal() throws Exception {
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Withdrawal/Withdrawal.java]: Rename in verb-noun format

File: platform/src/main/java/platform/pageobjects/Withdrawal/Withdrawal.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+	public Withdrawal(WebDriver driver) {
+		this.driver = driver;
+		PageFactory.initElements(driver, this);
+	}
+
+	@FindBy(xpath = "//span[@class='checkmark']")
+	public WebElement withdrawalCheck;
+
+	@FindBy(xpath = "//a[@class='button button--gray button--medium button--coupon-back']//span[contains(text(),'戻る')]")
```
Comment (csonparote) [inline platform/src/main/java/platform/pageobjects/Withdrawal/Withdrawal.java]: Describe web element in name
Ex: WebElement backButton;

File: platform/src/test/java/platformfeatures/PF_Withdrawal.feature [hunk 3]
```diff
@@ -0,0 +1,60 @@
+
+#Author: Sahaj Balgunde
+#Keywords Summary : Withdawal Test with Checkbox and Without Checkbox
+
+
+
+Feature: Withdraw user account
+@FunctionalTest
```
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_Withdrawal.feature]: Write exclusive tag for scenario

File: platform/src/test/java/platformfeatures/PF_Withdrawal.feature [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+#Given User is on Withdrawal Page
+#When User clicks on accept checkbox
+#Then the check box is clickable
+
+
+#Scenario: User is withdrawing the account
+
+Given Open Browser and URL <browser>
+Then User is on Withdrawal Page ticks the checkbox and withdraw account
```
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_Withdrawal.feature]: Include user actions in When statement
Validate test in Then statement

File: platform/src/test/java/platformfeatures/PF_Withdrawal.feature [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+Then Account gets deleted
```
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_Withdrawal.feature]: Mulitple Then not necessary
you can add step with And

Ex:
Given pre requisite
And another pre requisite
When user performs Action 
And perform another action
Then validate test
And validate another step

File: platform/src/test/java/platformfeatures/PF_Withdrawal.feature [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+Then Browser is closed
+
+
+Examples:
+|browser|
+|chrome	|
+
+#Scenario: User is withdrawing the account
```
Comment (csonparote) [inline platform/src/test/java/platformfeatures/PF_Withdrawal.feature]: Please remove commented code block

File: platform/src/test/java/platformstepdefinition/PF_Withdrawal.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+	public WebDriver driver;
+
+	@Given("^Open Browser and URL (.+)$")
+	public void open_Browser_and_URL(String browser) throws Throwable {
+		driver = openbrowser(browser);
+	}
+
+	@Then("^User is on Withdrawal Page ticks the checkbox and withdraw account$")
+	public void user_is_on_Withdrawal_Page_ticks_the_checkbox_and_withdraw_account(String browser) throws Throwable {
+		LoginPage loginToPlatform = new LoginPage(driver);
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Withdrawal.java]: New feature step can be created for login

File: platform/src/test/java/platformstepdefinition/PF_Withdrawal.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+		withdrawal.accountWithdrawal();
+
+		System.out.println("Account has been deleted.");
+
+	}
+
+	@Then("^Account gets deleted$")
+	public void Account_gets_deleted(String browser) throws Throwable {
+
+		System.out.println("Account has been deleted.");
```
Comment (csonparote) [inline platform/src/test/java/platformstepdefinition/PF_Withdrawal.java]: How are you validating account deletion?

**Review by csonparote — CHANGES_REQUESTED** [review]

File: platform/src/test/java/platformstepdefinition/PF_Withdrawal.java (hunk 8 above)
Comment (sahajisy) [reply platform/src/test/java/platformstepdefinition/PF_Withdrawal.java]: @csonparote On 2nd Then statement, account gets withdrawn. Need help for 3rd Then Statement.

Comment (csonparote) [reply platform/src/test/java/platformstepdefinition/PF_Withdrawal.java]: Try to login with old credentials
If error message then account deletion is validated

**Comment by sahajisy** [comment]
Updated Pageobject, feature file and stepdefinition, there seems to be no errors but unable to run the testcase. Help needed regarding to it.

File: platform/src/main/java/platform/pageobjects/Withdrawal/Withdrawal.java (hunk 1 above)
Comment (sahajisy) [reply platform/src/main/java/platform/pageobjects/Withdrawal/Withdrawal.java]: Updated as requested.

File: platform/src/test/java/platformfeatures/PF_Withdrawal.feature (hunk 3 above)
Comment (sahajisy) [reply platform/src/test/java/platformfeatures/PF_Withdrawal.feature]: Updated as requested.

File: platform/src/test/java/platformfeatures/PF_Withdrawal.feature (hunk 4 above)
Comment (sahajisy) [reply platform/src/test/java/platformfeatures/PF_Withdrawal.feature]: Updated as requested.

File: platform/src/test/java/platformfeatures/PF_Withdrawal.feature (hunk 5 above)
Comment (sahajisy) [reply platform/src/test/java/platformfeatures/PF_Withdrawal.feature]: Updated as requested.

File: platform/src/test/java/platformfeatures/PF_Withdrawal.feature (hunk 6 above)
Comment (sahajisy) [reply platform/src/test/java/platformfeatures/PF_Withdrawal.feature]: Updated as requested.

File: platform/src/test/java/platformstepdefinition/PF_Withdrawal.java (hunk 8 above)
Comment (sahajisy) [reply platform/src/test/java/platformstepdefinition/PF_Withdrawal.java]: Updated as requested.

**Review by csonparote — APPROVED** [review]
Approved . Plz merge

**Comment by sahajisy** [comment]
@csonparote san, Can we create new function of login for Withdrawing account on LoginPage. java? 
It can be helpful because main account we are using in automation will not be deleted.

**Pull request merged**